tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1178,6 +1213,139 @@ bool CorProfiler::ProfilerAssemblyIsLoadedIntoAppDomain(AppDomainID app_domain_i
            managed_profiler_loaded_app_domains.find(app_domain_id) != managed_profiler_loaded_app_domains.end();
 }
 
+HRESULT CorProfiler::RewriteForDistributedTracing(const ModuleMetadata& module_metadata, ModuleID module_id)
+{
+    HRESULT hr = S_OK;
+
+    if (IsDebugEnabled())
+    {
+        Logger::Info("pcbPublicKey: ", managed_profiler_assembly_property.pcbPublicKey);
+        Logger::Info("pcbPublicKey: ", HexStr(managed_profiler_assembly_property.ppbPublicKey,
+                                              managed_profiler_assembly_property.pcbPublicKey));
+        Logger::Info("pcbPublicKey: ");
+        const auto ppbPublicKey = (BYTE*) managed_profiler_assembly_property.ppbPublicKey;
+        for (auto i = 0; i < managed_profiler_assembly_property.pcbPublicKey; i++)
+        {
+            Logger::Info(" -> ", (int) ppbPublicKey[i]);
+        }
+        Logger::Info("szName: ", managed_profiler_assembly_property.szName);
+
+        Logger::Info("Metadata.cbLocale: ", managed_profiler_assembly_property.pMetaData.cbLocale);
+        Logger::Info("Metadata.szLocale: ", managed_profiler_assembly_property.pMetaData.szLocale);
+
+        if (managed_profiler_assembly_property.pMetaData.rOS != nullptr)
+        {
+            Logger::Info("Metadata.rOS.dwOSMajorVersion: ",
+                         managed_profiler_assembly_property.pMetaData.rOS->dwOSMajorVersion);
+            Logger::Info("Metadata.rOS.dwOSMinorVersion: ",
+                         managed_profiler_assembly_property.pMetaData.rOS->dwOSMinorVersion);
+            Logger::Info("Metadata.rOS.dwOSPlatformId: ",
+                         managed_profiler_assembly_property.pMetaData.rOS->dwOSPlatformId);
+        }
+
+        Logger::Info("Metadata.usBuildNumber: ", managed_profiler_assembly_property.pMetaData.usBuildNumber);
+        Logger::Info("Metadata.usMajorVersion: ", managed_profiler_assembly_property.pMetaData.usMajorVersion);
+        Logger::Info("Metadata.usMinorVersion: ", managed_profiler_assembly_property.pMetaData.usMinorVersion);
+        Logger::Info("Metadata.usRevisionNumber: ", managed_profiler_assembly_property.pMetaData.usRevisionNumber);
+
+        Logger::Info("pulHashAlgId: ", managed_profiler_assembly_property.pulHashAlgId);
+        Logger::Info("sizeof(pulHashAlgId): ", sizeof(managed_profiler_assembly_property.pulHashAlgId));
+        Logger::Info("assemblyFlags: ", managed_profiler_assembly_property.assemblyFlags);
+    }
+
+    //
+    // *** Get DistributedTracer TypeDef
+    //
+    mdTypeDef distributedTracerTypeDef;
+    hr = module_metadata.metadata_import->FindTypeDefByName(WStr("Datadog.Trace.ClrProfiler.DistributedTracer"),
+                                                            mdTokenNil, &distributedTracerTypeDef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error Rewriting for Distributed Tracing on getting DistributedTracer TypeDef");
+        return hr;
+    }
```

Comment:
What's the failure mode here if we fail to rewrite? I'll figure it out further down the PR, just want to make sure we essentially fail in the same way as before (disconnected traces) and not with null refs or similar